adafruit_avrprog.py

```diff
@@ -317,7 +325,7 @@ def read_signature(self):
             sig.append(self._transaction((0x30, 0, i, 0))[2])
         return sig
 
-    def read(self, addr, read_buffer):
+    def read(self, addr: int, read_buffer: bytes) -> None:
```

Comment:
Since we're writing things into what's provided into `read_buffer`, we'll need to use a type that can support writing using the indices.  In this case, let's use `bytearray`, which supports that.  Sometimes we can use an even less specific type, but given how it's used, I do think that only `bytearray` will work.